pynetdicom/service_class.py

```diff
@@ -1457,6 +1457,10 @@ def SCP(self, req, context):
         rsp = self.validate_status(rsp_status, rsp)
         self.dimse.send_msg(rsp, context.context_id)
 
+        # Cleanup
+        if isinstance(req.DataSet, FileIO):
+            req.DataSet.close()
```

Comment:
There's a return statement a couple of lines up which would leave the FileIO open.

What would happen if an exception occurred in the SCP?